server/base/src/main/java/org/apache/accumulo/server/metadata/AsyncConditionalTabletsMutatorImpl.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.server.metadata;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.function.BiConsumer;
+
+import org.apache.accumulo.core.dataImpl.KeyExtent;
+import org.apache.accumulo.core.metadata.schema.Ample;
+import org.apache.accumulo.server.ServerContext;
+
+public class AsyncConditionalTabletsMutatorImpl implements Ample.AsyncConditionalTabletsMutator {
+  private final BiConsumer<KeyExtent,Ample.ConditionalResult> resultsConsumer;
+  private final ExecutorService executor;
+  private Future<Map<KeyExtent,Ample.ConditionalResult>> backgroundProcessing = null;
+  private ConditionalTabletsMutatorImpl bufferingMutator;
+  private final ServerContext context;
+  private long mutatedTablets = 0;
+
+  public static final int BATCH_SIZE = 1000;
+
+  AsyncConditionalTabletsMutatorImpl(ServerContext context,
+      BiConsumer<KeyExtent,Ample.ConditionalResult> resultsConsumer) {
+    this.resultsConsumer = Objects.requireNonNull(resultsConsumer);
+    this.bufferingMutator = new ConditionalTabletsMutatorImpl(context);
+    this.context = context;
+    this.executor = Executors.newSingleThreadExecutor();
```

Comment:
The background task uses a conditional writer which is itself multithreaded and by defaults uses three threads.  This made me realize that we made need to explore making that configurable in ample so that in the case where metadata tablets span multiple tservers that the conditional writer could write to more in parallel.

Reply:
> We could use a cached pool or a bounded pool if we wanted to limit the number of concurrent fate operations. (I'm not sure if the number of operations is already bound elsewhere).

There is a thread pool for FATE operations and its size is configurable.

Reply:
This little background thread would be a good use case for the new light weight threads introduced in JDK 21 as all its really doing is waiting on the results from the conditional writer.